cmd/buildah/containers.go

```diff
@@ -64,18 +72,57 @@ func containersCmd(c *cli.Context) error {
 			fmt.Printf("%-64s %-64s %-10s %s\n", "CONTAINER ID", "IMAGE ID", "IMAGE NAME", "CONTAINER NAME")
 		}
 	}
-	for _, builder := range builders {
-		if builder.FromImage == "" {
-			builder.FromImage = buildah.BaseImageFakeName
-		}
+	list := func(containerID, imageID, image, container string) {
 		if quiet {
-			fmt.Printf("%s\n", builder.ContainerID)
+			fmt.Printf("%s\n", containerID)
 		} else {
 			if truncate {
-				fmt.Printf("%-12.12s %-12.12s %-10s %s\n", builder.ContainerID, builder.FromImageID, builder.FromImage, builder.Container)
+				fmt.Printf("%-12.12s %-12.12s %-10s %s\n", containerID, imageID, image, container)
+			} else {
+				fmt.Printf("%-64s %-64s %-10s %s\n", containerID, imageID, image, container)
+			}
+		}
+	}
+	seenContainers := make(map[string]struct{})
+	for _, builder := range builders {
```

Comment:
Just wondering if it makes sense to do the list of just buildah containers (lines ~75->95) under an "if !all" statement then make the current "if all" an else for that and remove the seenImages functionality.  I'm not sure how painful the lookups could be under a big load.